components/tools/OmeroJava/test/integration/gateway/GatewayUsageTest.java

```diff
@@ -56,6 +60,24 @@ public void testLoginWithCredentials()
         gw.disconnect();
     }
 
+    /**
+     * Check logging in using a session ID.
+     * @throws DSOutOfServiceException unexpected
+     */
+    @Test
+    public void testLoginWithCredentialsSessionId()
+            throws DSOutOfServiceException {
+        final omero.client client =  new omero.client();
```

Comment:
Looking at this test, this raises a general design question to me. From  the point of view of a consumer of this libary, is there any benefit into migrating this logic to the gateway itself. For instance,  define a  simpler

```
 public LoginCredentials(String sessionId)
```

which would built an `omero.client()` using the ice.config set by envvar and pass the requires properties to  the constructor? Or should this logic be migrated away from `LoginCredentials` by design?